In [52]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

from sklearn.impute import SimpleImputer

import python_code.Scripts as sc
import python_code.Reference as ref

In [3]:
snap07 = pd.read_spss('./data/qc_pub_fy2007.sav')

In [4]:
snap07.head(5)

FSAFIL1  FSAFIL2  FSAFIL3  FSAFIL4  FSAFIL5  FSAFIL6  FSAFIL7  FSAFIL8  \
0      1.0      1.0      1.0      NaN      NaN      NaN      NaN      NaN   
1      1.0      1.0      1.0      1.0      1.0      NaN      NaN      NaN   
2      1.0      1.0      NaN      NaN      NaN      NaN      NaN      NaN   
3      1.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
4      1.0      1.0      1.0      1.0      1.0      NaN      NaN      NaN   

   FSAFIL9  FSAFIL10  ...  NDISCA7  NDISCA8  NDISCA9  NDISCA10  NDISCA11  \
0      NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
1      NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
2      NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
3      NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
4      NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   

   NDISCA12  NDISCA13  NDISCA14  NDISCA15  NDISCA16  
0       NaN       NaN       NaN       NaN       NaN  
1       NaN       NaN       NaN       NaN       NaN  
2       NaN       NaN       NaN       NaN       NaN  
3       NaN       NaN       NaN       NaN       NaN  
4       NaN       NaN       NaN       NaN       NaN  

[5 rows x 769 columns]

In [12]:
snap07.shape

(47469, 769)

In [13]:
snap07.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 47469 entries, 0 to 47468
Columns: 769 entries, FSAFIL1 to NDISCA16
dtypes: float64(769)
memory usage: 278.5 MB


In [14]:
snap07.isnull().sum()

FSAFIL1         0
FSAFIL2     20739
FSAFIL3     29470
FSAFIL4     36884
FSAFIL5     42301
            ...  
NDISCA12    47455
NDISCA13    47463
NDISCA14    47466
NDISCA15    47469
NDISCA16    47469
Length: 769, dtype: int64

## *2017 Data*

In [34]:
snap17 = pd.read_csv('./data/qc_pub_fy2017.csv')

In [35]:
snap17.head(5)

FSAFIL1  FSAFIL2  FSAFIL3  FSAFIL4  FSAFIL5  FSAFIL6  FSAFIL7  FSAFIL8  \
0        1      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
1        1      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
2        1      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
3        1      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
4        1      1.0      NaN      NaN      NaN      NaN      NaN      NaN   

   FSAFIL9  FSAFIL10  ...  WORK12  WORK13  WORK14  WORK15  WORK16  FSNONCIT  \
0      NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN         0   
1      NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN         0   
2      NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN         0   
3      NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN         0   
4      NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN         0   

   FSDIS  FSELDER  FSKID    STATENAME  
0      1        0      0  Connecticut  
1      0        0      0  Connecticut  
2      0        0      0  Connecticut  
3      1        0      0  Connecticut  
4      0        1      0  Connecticut  

[5 rows x 814 columns]

In [36]:
snap17.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45530 entries, 0 to 45529
Columns: 814 entries, FSAFIL1 to STATENAME
dtypes: float64(368), int64(445), object(1)
memory usage: 282.8+ MB


In [37]:
snap17.isnull().sum()

FSAFIL1          0
FSAFIL2      22368
FSAFIL3      30458
FSAFIL4      36350
FSAFIL5      40801
             ...  
FSNONCIT         0
FSDIS            0
FSELDER          0
FSKID            0
STATENAME        0
Length: 814, dtype: int64

## *Target Variable* (what we will be predicting = Y variable)

In [10]:
# 2007: 1 = Eligible, 2 = Not eligible
snap07['CAT_ELIG'].value_counts()

CAT_ELIG
0.0    25134
1.0    22335
Name: count, dtype: int64

In [38]:
# 2017: 0 = Not eligible, 1 = Reported eligible, 2= Recorded eligible
snap17['CAT_ELIG'].value_counts()

CAT_ELIG
1    36223
0     7012
2     2295
Name: count, dtype: int64

#### **Observations:** I am going to change the target variable in both datasets to a dictionary of:
- 0 = Not Eligible
- 1 = Eligible

In [39]:
snap07['CAT_ELIG'] = snap07['CAT_ELIG'].replace(2,0)
snap17['CAT_ELIG'] = snap17['CAT_ELIG'].replace(2,1)

In [12]:
snap07['CAT_ELIG'].value_counts()

CAT_ELIG
0.0    25134
1.0    22335
Name: count, dtype: int64

In [40]:
snap17['CAT_ELIG'].value_counts()

CAT_ELIG
1    38518
0     7012
Name: count, dtype: int64

In [41]:
print(f'2017 dataset: {snap17.shape} VS 2007 dataset: {snap07.shape}')

2017 dataset: (45530, 814) VS 2007 dataset: (47469, 769)


#### **Observations:** Less people nationally applied for SNAP benefits in 2017 as opposed to 2007. This probably had to do with stronger national economic factors such as employment opportunities. Also, 45 columns of features were added to the dataset.

## *Reference Tables*

In [42]:
import python_code.Reference as ref
print(f'Unit Demo, 2007 is {ref.unit07_demo.shape[0]} and 2017 is {ref.unit17_demo.shape[0]}')
print(f'Unit Assets, 2007 is {ref.unit07_assets.shape[0]} and 2017 is {ref.unit17_assets.shape[0]}')
print(f'Unit ExDed, 2007 is {ref.unit07_exded.shape[0]} and 2017 is {ref.unit17_exded.shape[0]}')
print(f'Unit Inc, 2007 is {ref.unit07_inc.shape[0]} and 2017 is {ref.unit17_inc.shape[0]}')
print(f'Pers Char, 2007 is {ref.per07_char.shape[0]} and 2017 is {ref.per17_char.shape[0]}')
print(f'Pers Inc, 2007 is {ref.per07_inc.shape[0]} and 2017 is {ref.per17_inc.shape[0]}')

Unit Demo, 2007 is 22 and 2017 is 31
Unit Assets, 2007 is 7 and 2017 is 11
Unit ExDed, 2007 is 27 and 2017 is 27
Unit Inc, 2007 is 26 and 2017 is 27
Pers Char, 2007 is 14 and 2017 is 17
Pers Inc, 2007 is 20 and 2017 is 21


#### **Observations:** From 2007 to 2017, every category obtained more columns except for Expenses and Deductables.

# Extracting State Data

## *New Mexico*

In [16]:
#save New Mexico records
#import pandas as pd
#snap07 = pd.read_spss('./data/qc_pub_fy2007.sav')
nm07 = snap07.loc[snap07['STATE'] == 35].astype('float64')
nm07_target = nm07['CAT_ELIG']
nm07.to_csv('./data/nm07.csv',index=None)

In [43]:
#snap17 = pd.read_csv('./data/qc_pub_fy2017.csv')
nm17 = snap17.loc[snap17['STATE'] == 35]
nm17_target = nm17['CAT_ELIG']
nm17 = nm17.drop(columns = ['STATENAME'])
nm17 = nm17.astype('float64')
nm17.to_csv('./data/nm17.csv',index=None)

In [23]:
#New Mexico dataframe
df_nm07 = pd.read_csv('./data/nm07.csv')
df_nm07

FSAFIL1  FSAFIL2  FSAFIL3  FSAFIL4  FSAFIL5  FSAFIL6  FSAFIL7  FSAFIL8  \
0         1.0      1.0      1.0      NaN      NaN      NaN      NaN      NaN   
1         1.0      1.0      1.0      1.0      1.0      1.0      NaN      NaN   
2         1.0      1.0      1.0      NaN      NaN      NaN      NaN      NaN   
3         1.0      1.0      1.0      1.0      NaN      NaN      NaN      NaN   
4         1.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
...       ...      ...      ...      ...      ...      ...      ...      ...   
1250      1.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
1251      1.0      1.0      NaN      NaN      NaN      NaN      NaN      NaN   
1252      1.0      1.0      NaN      NaN      NaN      NaN      NaN      NaN   
1253      1.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
1254      1.0      1.0      1.0      1.0      1.0      1.0      1.0      1.0   

      FSAFIL9  FSAFIL10  ...  NDISCA7  NDISCA8  NDISCA9  NDISCA10  NDISCA11  \
0         NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
1         NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
2         NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
3         NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
4         NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
...       ...       ...  ...      ...      ...      ...       ...       ...   
1250      NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
1251      NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
1252      NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
1253      NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
1254      1.0       NaN  ...      0.0      0.0      0.0       NaN       NaN   

      NDISCA12  NDISCA13  NDISCA14  NDISCA15  NDISCA16  
0          NaN       NaN       NaN       NaN       NaN  
1          NaN       NaN       NaN       NaN       NaN  
2          NaN       NaN       NaN       NaN       NaN  
3          NaN       NaN       NaN       NaN       NaN  
4          NaN       NaN       NaN       NaN       NaN  
...        ...       ...       ...       ...       ...  
1250       NaN       NaN       NaN       NaN       NaN  
1251       NaN       NaN       NaN       NaN       NaN  
1252       NaN       NaN       NaN       NaN       NaN  
1253       NaN       NaN       NaN       NaN       NaN  
1254       NaN       NaN       NaN       NaN       NaN  

[1255 rows x 769 columns]

In [44]:
#New Mexico dataframe
df_nm17 = pd.read_csv('./data/nm17.csv')
df_nm17

FSAFIL1  FSAFIL2  FSAFIL3  FSAFIL4  FSAFIL5  FSAFIL6  FSAFIL7  FSAFIL8  \
0        1.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
1        1.0      1.0      NaN      NaN      NaN      NaN      NaN      NaN   
2        1.0      1.0      NaN      NaN      NaN      NaN      NaN      NaN   
3        1.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
4        1.0      1.0      NaN      NaN      NaN      NaN      NaN      NaN   
..       ...      ...      ...      ...      ...      ...      ...      ...   
959      1.0      1.0      NaN      NaN      NaN      NaN      NaN      NaN   
960      1.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
961      1.0      1.0      1.0      1.0      1.0      NaN      NaN      NaN   
962      1.0      1.0      NaN      NaN      NaN      NaN      NaN      NaN   
963      1.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN   

     FSAFIL9  FSAFIL10  ...  WORK11  WORK12  WORK13  WORK14  WORK15  WORK16  \
0        NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
1        NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
2        NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
3        NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
4        NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
..       ...       ...  ...     ...     ...     ...     ...     ...     ...   
959      NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
960      NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
961      NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
962      NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
963      NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   

     FSNONCIT  FSDIS  FSELDER  FSKID  
0         0.0    0.0      0.0    0.0  
1         0.0    0.0      1.0    1.0  
2         0.0    0.0      0.0    1.0  
3         0.0    0.0      0.0    0.0  
4         0.0    1.0      0.0    1.0  
..        ...    ...      ...    ...  
959       0.0    0.0      0.0    1.0  
960       0.0    0.0      0.0    0.0  
961       0.0    0.0      0.0    1.0  
962       0.0    1.0      1.0    0.0  
963       0.0    0.0      0.0    0.0  

[964 rows x 813 columns]

In [20]:
df_nm07['CAT_ELIG'].value_counts()

CAT_ELIG
0.0    813
1.0    442
Name: count, dtype: int64

In [45]:
df_nm17['CAT_ELIG'].value_counts()

CAT_ELIG
1.0    964
Name: count, dtype: int64

## *Nebraska*

In [46]:
# Save Nebrasks records
ne07 = snap07.loc[snap07['STATE'] == 31].astype('float64')
ne07_target = ne07['CAT_ELIG']
ne07.to_csv('./data/ne07.csv',index=None)

In [47]:
ne17 = snap17.loc[snap17['STATE'] == 31]
ne17_target = ne17['CAT_ELIG']
ne17 = ne17.drop(columns = ['STATENAME'])
ne17 = ne17.astype('float64')
ne17.to_csv('./data/ne17.csv',index=None)

In [48]:
#Nebraska dataframe
df_ne07 = pd.read_csv('./data/ne07.csv')
df_ne07

FSAFIL1  FSAFIL2  FSAFIL3  FSAFIL4  FSAFIL5  FSAFIL6  FSAFIL7  FSAFIL8  \
0        1.0      1.0      1.0      NaN      NaN      NaN      NaN      NaN   
1        1.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
2        1.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
3        1.0      1.0      1.0      1.0      1.0      NaN      NaN      NaN   
4        1.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
..       ...      ...      ...      ...      ...      ...      ...      ...   
786      1.0      1.0      1.0      1.0      NaN      NaN      NaN      NaN   
787      1.0      1.0      1.0      NaN      NaN      NaN      NaN      NaN   
788      1.0      1.0      NaN      NaN      NaN      NaN      NaN      NaN   
789      1.0      1.0      NaN      NaN      NaN      NaN      NaN      NaN   
790      1.0      1.0      1.0      1.0      NaN      NaN      NaN      NaN   

     FSAFIL9  FSAFIL10  ...  NDISCA7  NDISCA8  NDISCA9  NDISCA10  NDISCA11  \
0        NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
1        NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
2        NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
3        NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
4        NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
..       ...       ...  ...      ...      ...      ...       ...       ...   
786      NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
787      NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
788      NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
789      NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   
790      NaN       NaN  ...      NaN      NaN      NaN       NaN       NaN   

     NDISCA12  NDISCA13  NDISCA14  NDISCA15  NDISCA16  
0         NaN       NaN       NaN       NaN       NaN  
1         NaN       NaN       NaN       NaN       NaN  
2         NaN       NaN       NaN       NaN       NaN  
3         NaN       NaN       NaN       NaN       NaN  
4         NaN       NaN       NaN       NaN       NaN  
..        ...       ...       ...       ...       ...  
786       NaN       NaN       NaN       NaN       NaN  
787       NaN       NaN       NaN       NaN       NaN  
788       NaN       NaN       NaN       NaN       NaN  
789       NaN       NaN       NaN       NaN       NaN  
790       NaN       NaN       NaN       NaN       NaN  

[791 rows x 769 columns]

In [49]:
df_ne17 = pd.read_csv('./data/ne17.csv')
df_ne17

FSAFIL1  FSAFIL2  FSAFIL3  FSAFIL4  FSAFIL5  FSAFIL6  FSAFIL7  FSAFIL8  \
0        1.0      1.0      NaN      NaN      NaN      NaN      NaN      NaN   
1        1.0      1.0      NaN      NaN      NaN      NaN      NaN      NaN   
2        1.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
3        1.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
4        1.0      1.0      NaN      NaN      NaN      NaN      NaN      NaN   
..       ...      ...      ...      ...      ...      ...      ...      ...   
889      1.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
890      1.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
891      1.0      1.0      NaN      NaN      NaN      NaN      NaN      NaN   
892      1.0      1.0      NaN      NaN      NaN      NaN      NaN      NaN   
893      1.0      1.0      NaN      NaN      NaN      NaN      NaN      NaN   

     FSAFIL9  FSAFIL10  ...  WORK11  WORK12  WORK13  WORK14  WORK15  WORK16  \
0        NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
1        NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
2        NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
3        NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
4        NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
..       ...       ...  ...     ...     ...     ...     ...     ...     ...   
889      NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
890      NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
891      NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
892      NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
893      NaN       NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   

     FSNONCIT  FSDIS  FSELDER  FSKID  
0         0.0    0.0      0.0    0.0  
1         0.0    1.0      0.0    0.0  
2         0.0    0.0      1.0    0.0  
3         0.0    1.0      0.0    0.0  
4         0.0    0.0      1.0    0.0  
..        ...    ...      ...    ...  
889       0.0    0.0      0.0    0.0  
890       0.0    0.0      0.0    0.0  
891       0.0    1.0      0.0    1.0  
892       0.0    0.0      0.0    1.0  
893       0.0    0.0      0.0    1.0  

[894 rows x 813 columns]

In [30]:
df_ne07['CAT_ELIG'].value_counts()

CAT_ELIG
0.0    505
1.0    286
Name: count, dtype: int64

In [50]:
df_ne17['CAT_ELIG'].value_counts()

CAT_ELIG
1.0    894
Name: count, dtype: int64

#### **Observations:** 
We can see there are 1255 people who applied for SNAP in New Mexico, opposed to 791 applications in Nebraska in 2007.

There were significantly more people in both states (Nebraska: 894, New Mexico: 964) on SNAP in 2017. Due to a dramatic change, I suspect that it was due to more being eligible because of a change in qualifications.

# Initial Snapshots of the Data

In [54]:
sc.plot_simple_features('FSNELDER','elderly',"Number of Elderly(over age 60)")

ValueError: 
\itNM07
^
ParseFatalException: Unknown symbol: \itNM, found '\'  (at char 0), (line:1, col:1)

Error in callback <function _draw_all_if_interactive at 0x0000022796091F80> (for post_execute):


ValueError: 
\itNM07
^
ParseFatalException: Unknown symbol: \itNM, found '\'  (at char 0), (line:1, col:1)

ValueError: 
\itNM07
^
ParseFatalException: Unknown symbol: \itNM, found '\'  (at char 0), (line:1, col:1)

<Figure size 1600x1000 with 4 Axes>